pjmedia/src/pjmedia-codec/ffmpeg_vid_codecs.c

```diff
@@ -1861,7 +1911,18 @@ static pj_status_t ffmpeg_codec_decode_whole(pjmedia_vid_codec *codec,
     avpacket.flags = 0;
 #endif
 
-#if LIBAVCODEC_VER_AT_LEAST(52,72)
+#if LIBAVCODEC_VER_AT_LEAST(58,137)
+    err = avcodec_send_packet(ff->dec_ctx, &avpacket);
+    if (err >= 0) {
+        err = avcodec_receive_frame(ff->dec_ctx, &avframe);
+        if (err == AVERROR(EAGAIN) || err == AVERROR_EOF)
```

Comment:
according to https://ffmpeg.org/doxygen/3.4/group__lavc__decoding.html#ga11e6542c4e66d3028668788a1a74217c:
"[AVERROR(EAGAIN)](https://ffmpeg.org/doxygen/3.4/group__lavu__error.html#gae4bb6f165973d09584e0ec0f335f69ca): output is not available in this state - user must try to send new input AVERROR_EOF: the decoder has been fully flushed, and there will be no more output frames"

so is it correct that got_picture is true for this case?

Reply:
I think you are right